db.go

```diff
@@ -634,7 +656,11 @@ func (db *DB) Exec(query interface{}, args ...interface{}) (sql.Result, error) {
 
 	start := time.Now()
 	argsConverted := argsConvert(args)
-	result, err := exec(db.GetContext(), db.DB, querystr, argsConverted...)
+	dbCtx := db.Context()
+	if !isEmptyContext(ctx) {
```

Comment:
For methods that take a `Context`, I think that context should be used as-is rather than conditionally replaced by a different context. Replacing the context in some cases could be surprising to the caller.

Reply:
Now that I think about it, I agree. If a caller wants to use the context on the `Executor`, they should just call the non-context version of the method. If they call the context version with an empty context, we should assume they explicitly want the context to be empty.